### ISSUE TITLE
Feature: stopwatch and timer

### ISSUE THREAD

**Issue by poetaman** [valebes/rsClock#3]
Stopwatch and timer would make this a killer CLI app! Thanks

**Comment by valebes** [comment]
Thanks for the suggestion.
Now I'm busy with work, later in August I'll work on something.

**Comment by Kyllingene** [comment]
That could be a laggy stopwatch, considering that it has to format and draw large block characters (and rsclock is [optimized for size](https://github.com/valebes/rsClock/blob/0a7cfabc54f56b73edbe32a1dcb4a44a551ed974/Cargo.toml#L24)). A timer would be handy, but I know from experience that notifying the user of a finished TUI timer is tricky.

You should try your hand at it yourself, and PR when you have it working; even if your PR doesn't get accepted, you have your own version with the functionality you want!

**Comment by poetaman** [comment]
@Kyllingene I wrote this: https://github.com/poetaman/arttime for my needs. It has cross-platform desktop notifications. A stopwatch need not update screen every 10 ms. Are you maintaining this project?

**Comment by Kyllingene** [comment]
No, simply giving friendly suggestions. I quickly tested what it might look like if rsClock updated rapidly, and I stand corrected; it worked fine (on Alacritty, Arch Linux).

My one remaining concern is desktop notifications. `arttime` is written in zsh (and only supports environments with `notify-send`), so unless maintaining cross-platform `Command`s sounds fun to valebes, that solution isn't portable. [`notify-rust`](https://crates.io/crates/notify-rust) seems promising (it supports Linux, Mac, and even Windows), if it's kept to a bare minimum (summary and body). I think the reason I dislike this option is just because it's adding several layers of complexity (separate modes of operation, `dbus` concerns); however, that's entirely up to valebes.

**Comment by valebes** [comment]
Probably adding notification management would be a "little too much"(as said by @Kyllingene), especially in terms of adding complexity to the entire code just for one (secondary) feature. 
Maybe add as a feature the option to set an alarm clock? But even there it would seem unnecessary to me to manage with the notification system.
I leave the issue open for possibly future updates.
